models/members.js

```diff
@@ -115,9 +115,40 @@ const addArchiveRoleToMembers = async (userId) => {
   }
 };
 
+const updateRoles = async (userId, newRoles) => {
+  try {
+    const userDoc = await userModel.doc(userId).get();
+    const user = userDoc.data();
```

Comment:
Fetching the user again here

Reply:
You asked question that why I fetch user data again?? or I have to fetch like this `const user = await fetchUser({ userId });`

Reply:
fixed